libs/wingsdk/src/target-tf-gcp/table.ts

```diff
@@ -0,0 +1,102 @@
+import { Construct } from "constructs";
+import { App } from "./app";
+import {
+  BigtableInstance,
+  BigtableInstanceCluster,
+  BigtableInstanceClusterAutoscalingConfig,
+  BigtableInstanceConfig,
+} from "../.gen/providers/google/bigtable-instance";
+import {
+  BigtableTable,
+  BigtableTableConfig,
+  BigtableTableColumnFamily,
+} from "../.gen/providers/google/bigtable-table";
+import * as ex from "../ex";
+import {
+  ResourceNames,
+  NameOptions,
+  CaseConventions,
+} from "../shared/resource-names";
+import { IInflightHost, Json } from "../std";
+
+const TABLE_NAME_OPTS: NameOptions = {
+  maxLen: 22,
+  disallowedRegex: /[a-z0-9\-\.\_]+/g,
+  sep: "A",
+};
+
+const INSTANCE_NAME_OPTS: NameOptions = {
+  maxLen: 22,
+  disallowedRegex: /[a-z0-9\-\.\_]+/g,
+  sep: "A",
+  case: CaseConventions.LOWERCASE,
+};
+
+/**
+ * GCP implementation of `ex.Table`.
+ *
+ * @inflight `@winglang/sdk.ex.ITableClient`
+ */
+export class Table extends ex.Table {
+  constructor(scope: Construct, id: string, props: ex.TableProps = {}) {
+    super(scope, id, props);
+
+    const app = App.of(this) as App;
+
+    const tableName = ResourceNames.generateName(this, TABLE_NAME_OPTS);
+    const instanceName = ResourceNames.generateName(this, INSTANCE_NAME_OPTS);
+
+    const columnsFamily: BigtableTableColumnFamily[] = [];
+    for (let key in this.columns) {
+      columnsFamily.push({ family: key });
+    }
+
+    let clusterId = props.clusterId;
+    if (!props.clusterId) {
+      clusterId = "default";
+    }
+
+    const autoscalingConfig: BigtableInstanceClusterAutoscalingConfig = {
+      minNodes: props.minNodes ?? 1,
```

Comment:
I believe it is not the best way to specify props, since these props are needed only in GCP, but are available on more 'global' ex.Table. I am not sure how I should handle it.

Reply:
Good question! Yeah - it's not uncommon for clouds to have extra parameters on some of their resources. Options like `minNodes` and `maxNodes` are typically considered "non-functional" aspects of the resource API because they represent operational concerns that are usually independent from the business logic you are performing with the resource.

For the Wing SDK, the best practice right now is to set a reasonable default (usually a good starting point is whatever defaults the cloud provider recommends when you're configuring the resource through their web console). Then, if anyone needs different values, they can be overridden using a [compiler plugin](https://www.winglang.io/docs/tools/compiler-plugins). In the future, there might be additional ways to customize these options within the Wing language by casting `cloud.Table` into `gcp.Table` or other mechanics.